components/canvas/hooks/useCopyPaste.tsx

```diff
@@ -0,0 +1,56 @@
+import { useStoreDispatch } from "@/hooks/storeHooks";
+import { useEffect } from "react";
+
+export const useCopyPaste = (
+  target: any,
+  action: (target: any) => void,
+  validator?: (target: any) => void
+) => {
+  const dispatch = useStoreDispatch();
+
+  const copyToClipboard = async (target: any) => {
+    try {
+      const valid = validator ? validator(target) : true;
+      if (valid) {
+        await navigator.clipboard.writeText(JSON.stringify(target));
+        dispatch.setSnackMessage("Copied 📋");
+      }
+    } catch (error) {
+      if (error instanceof Error) {
+        console.error(error.message);
```

Comment:
Should an error message be displayed to the user or logged to Application Insights? I don't think any regular user will go to the console in dev tools.